### DISCUSSION TITLE
[python-agent] the agent does not work

### DISCUSSION THREAD

**Discussion by little-hang** [apache/skywalking#8440] [Q&A] [answered]
First：my python code as follows:
`
import json
from flask import Flask, request
from skywalking import agent, config

config.init(collector_address='127.0.0.1:12800', service_name='service-B')  # collector 换成对应的skywalking OAP的地址
config.flask_collect_http_params = True # flask接收到的http参数也保存
agent.start()
app = Flask(__name__)

@app.route('/is_adult', methods=['POST'])
def delete_regress_tasks():
    """
    判断是否是成年人   
    return: 
    """
    input_param = json.loads(request.get_data()) 
    age = input_param['age']
    is_adult = True if age >= 18 else False

    return json.dumps({
        'is_adult': is_adult
    })

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=8884, debug=True)`

Then：I start my python server,the log as follows:

![image](https://user-images.githubusercontent.com/24517540/149761013-24c41d50-1cb8-4b69-ad20-eaa2cbb62fd7.png)

**Comment by kezhenxu94** [comment, marked as answer]
You are using default protocol (gRPC) and connecting to the HTTP port (12800), please read the doc first

**Reply by little-hang** [reply]
Thanks。The python Agent doc give the opposite port。Actually，12800 means grpc, 11800 means http。
When I pip install apache-skywalking[http]，and use collector_address=***:11800， it's OK

**Reply by wu-sheng** [reply]
Python supports to report through gRPC or HTTP. You need to choose the right one.

**Reply by kezhenxu94** [reply]
> Actually，12800 means grpc, 11800 means http。

This is wrong. 12800 means http, 11800 means gRPC.

> When I pip install apache-skywalking[http]，and use collector_address=***:11800， it's OK

If you modify `collector_address=***:11800`, you don't need `pip install apache-skywalking[http]`, you only need `pip install apache-skywalking`, this enables gRPC protocol by default, but if you use `pip install apache-skywalking[http]` you installed unused dependencies for http protocol (do no harm though).

**Comment by deepankar-gupta** [comment]
hello i have a doubt.
python code:
```
from skywalking import agent, config

# Initialize the agent
config.init(collector_address='skywalking-skywalking-oap-1:11800', service_name='sql')
agent.start()

```

i get an error: 

raise KeyError(f'Invalid configuration option {key}')
KeyError: 'Invalid configuration option collector_address'


please can you guide me what is the issue?

**Reply by wu-sheng** [reply]
You should follow the docs,

https://skywalking.apache.org/docs/skywalking-python/latest/en/setup/configuration/#agent-core-configuration-options

Your key is not expected.
